### DISCUSSION TITLE
Option for mode "on"

### DISCUSSION THREAD

**Discussion by oscarsan1** [echavet/MitsubishiCN105ESPHome#161] [Q&A] [answered]
There no option for mode power "on"
      mode: [COOL, HEAT, FAN_ONLY, DRY]

I've seen de code that mode "on" exist in every mode command: 

void CN105Climate::controlMode() {
    switch (this->mode) {
    case climate::CLIMATE_MODE_COOL:
        ESP_LOGI("control", "changing mode to COOL");
        this->setModeSetting("COOL");
        this->setPowerSetting("ON");

Can be mode power "on" be an option. This power on in last mode?

**Comment by echavet** [comment, marked as answer]
Hello @oscarsan1 
yes it is a mitsubishi's feature but not a home assistant climate component one.
So theoretically it should be possible to test an external attribute defined in your yaml which would manually call the setPowerSetting method. But the method is protected in the code, and the implementation finally doesn't support setting the power ON alone. A mode must be provided.
Of course, if you are a developper, you can fork the repository and try to change this behavior which wouldn't be a difficult task. But I don't know how the climate unit would respond to this.
Another lead is to make an automation. Look here: #124
